pwndbg/heap/ptmalloc.py

```diff
@@ -312,6 +312,49 @@ def is_top_chunk(self):
         return self._is_top_chunk
 
 
+class Heap:
+    def __init__(self, addr=None):
+        allocator = pwndbg.heap.current
+        if addr is not None:
+            try:
+                ar_ptr = allocator.get_heap(addr)["ar_ptr"]
+                ar_ptr.fetch_lazy()
+            except Exception:
+                ar_ptr = None
+
+            if ar_ptr is not None and ar_ptr in (ar.address for ar in allocator.arenas):
+                self.arena = Arena(ar_ptr)
+            else:
+                self.arena = Arena(allocator.main_arena.address)
+        else:
+            self.arena = Arena(allocator.get_arena().address)
```

Comment:
Can we add some comments here to describe the implementation choices we do here?
- Why do we have 3 different cases for choosing/instantiating arenas?
- When can we fail to obtain `allocator.get_heap(addr)["ar_ptr"]`?
- Any details about finding a "non-main arena"

Reply:
I agree this is opaque (gimme a break I wrote this with Serenade 😂)
It's existing logic that again I wasn't keen on messing with too much in case it caused issues.
I intend to change a lot of this and consolidate all the redundant code in one place shortly, at which point I'll ensure everything is appropriately commented👍